### PR TITLE
[12.0][IMP] account_financial_risk sale_financial_risk: Performance. Not store fields. Links to values records. Multi-company. Multi-currency.

### PR DESCRIPTION
**Pull request by carlosdauden**
***_financial_risk:** (account_financial_risk, account_payment_return_financial_risk sale_financial_risk)
 - Optimize code and change stored computed fields to not store to improve multi-company support
 - Improve multi-currency support
 - Convert risk amount fields to clickable link that shows traceability of amount origin
 - New pivot views to risk amount traceability
 - Simplify class style applied on risk fields
 - Migration script to remove old stored computed fields
 - Improve tests to cover new functionallity
 - Update translation files

**account_financial_risk:**
 - Don't block refund invoice validation when partner has risk exception
 - Allow search partners by risk exception field
 - Remove obsolete cron

**sale_financial_risk**
 - Create related store commercial_partner_id field in sale order line to simplify computation
 - Rename amt_to_invoice field to risk_amount in sale order line
 - Hook and migration scripts to reduce new fields computing time

@Tecnativa TT23765

financial_risk_click

### CODE REVIEW

File: sale_financial_risk/hooks.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    of the module on an existing Odoo instance.
+    """
+    create_commercial_partner_id_column(cr)
+
+
+def create_commercial_partner_id_column(cr):
+    cr.execute(
+        """
+        ALTER TABLE sale_order_line
+        ADD COLUMN IF NOT EXISTS commercial_partner_id int4;
```
Comment (chienandalu) [inline sale_financial_risk/hooks.py]: Looks like the instances of this repo are running on postgres 9.5, wich doesn't support this syntax

File: sale_financial_risk/views/sale_financial_risk_view.xml [hunk 2]
```diff
@@ -0,0 +1,16 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+
+    <record id="financial_risk_sale_order_line_pivot_view" model="ir.ui.view">
+        <field name="name">Account financial risk sale pivot view</field>
+        <field name="model">sale.order.line</field>
+        <field name="arch" type="xml">
+            <pivot string="Sale lines">
+                <field name="order_partner_id" type="row"/>
+<!--                <field name="date_order" interval="month" type="col"/>-->
```
Comment (chienandalu) [inline sale_financial_risk/views/sale_financial_risk_view.xml]: Is this on purpose?

**Review by chienandalu — COMMENTED** [review]
¿No readme updates?

File: account_financial_risk/data/account_financial_risk_data.xml [hunk 3]
```diff
@@ -1,17 +0,0 @@
-<?xml version="1.0"?>
-<!-- Copyright 2016-2018 Tecnativa - Carlos Dauden
-     License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl). -->
-<odoo noupdate="1">
-
-    <record id="ir_cron_due_invoice_every_day" model="ir.cron">
```
Comment (pedrobaeza) [inline account_financial_risk/data/account_financial_risk_data.xml:6]: You have to delete this record in the migration script. You can use `delete_records_safely_by_xml_id` of openupgradelib

File: sale_financial_risk/models/sale.py [hunk 4]
```diff
@@ -37,54 +41,49 @@ def action_confirm(self):
...
 
+    # TODO: Rename to risk_amount
     amt_to_invoice = fields.Monetary(
         string="Amount To Invoice",
         compute='_compute_amount_to_invoice',
         compute_sudo=True,
         store=True
     )
+    # TODO: Analyze performance vs order_id.partner_id.commercial_partner_id
+    commercial_partner_id = fields.Many2one(
```
Comment (pedrobaeza) [inline sale_financial_risk/models/sale.py:51]: index=True?

File: sale_financial_risk/models/sale.py [hunk 5]
```diff
@@ -37,54 +41,49 @@ def action_confirm(self):
 class SaleOrderLine(models.Model):
     _inherit = 'sale.order.line'
 
+    # TODO: Rename to risk_amount
```
Comment (pedrobaeza) [inline sale_financial_risk/models/sale.py]: As you are putting migration scripts, please do already the renaming

**Review by pedrobaeza — DISMISSED** [review]
Is Odoo removing automatically SQL columns of the previously stored computed fields? If not, you need to add that in the migration scripts.

I also find that there's no commit separation between each feature, and it even seems to include fixes that in the computation that are independent from non stored or link. Please try to split things for better review. If not, it's very difficult to hunt any possible problem

**Review by chienandalu — COMMENTED** [review]
Very cool :) Tested on runbot :+1: Missing a little update in the readme with the new feature.

**Review by rafaelbn — APPROVED** [review]
Functionally tested in runbot :+1:  thanks

**Comment by pedrobaeza** [comment]
Are you going to perform the commit separation?

**Comment by carlosdauden** [comment]
No, I'm working in other comments

**Review by sergio-teruel — APPROVED** [review]

**Comment by carlosdauden** [comment]
Changes done

**Review by chienandalu — APPROVED** [review]

**Pull request merged**